### PR TITLE
Better hypersurface support

### PR DESCRIPTION
**Pull request by ghislainb**

### CODE REVIEW

File: src/cellfunction/cellfunction.jl [hunk 1]
```diff
@@ -614,6 +527,44 @@ function LazyOperators.materialize(::CellNormal, cPoint::CellPoint{ReferenceDoma
...
+    fInfo = get_faceinfo(fPoint)
+    cInfo = get_cellinfo_p(fInfo)
+    cnodes = nodes(cInfo)
+    ctype = celltype(cInfo)
+    ξcell = get_coords(side_p(fPoint))
+
+    return cell_normal(ctype, cnodes, ξcell)
+end
+
+_tangeantial_projector(ν) = I - (ν ⊗ ν)
```
Comment (bmxam) [inline src/cellfunction/cellfunction.jl:565]: typo, should be "tangential" not "tangeantial"

File: src/cellfunction/cellfunction.jl [hunk 2]
```diff
@@ -519,53 +454,31 @@ function LazyOperators.materialize(
...
 
 function LazyOperators.materialize(
-    ::CoplanarRotation,
+    cr::CoplanarRotation,
     sideFacePoint::Side⁺{Nothing, <:Tuple{<:FacePoint}},
 )
-    cnodes_n, cnodes_p, ctype_n, ctype_p, ξ_n, ξ_p = _unpack_face_point(sideFacePoint)
-    return _coplanar_rotation(cnodes_p, cnodes_n, ctype_p, ctype_n, ξ_p, ξ_n)
+    fPoint, = get_args(sideFacePoint)
+    materialize(cr, Side⁻(opposite_side(fPoint)))
```
Comment (bmxam) [inline src/cellfunction/cellfunction.jl:471]: the necessity of `opposite_side` is just for dispatch problem write? Not for numerical reason?

File: src/mesh/domain.jl [hunk 3]
```diff
@@ -297,6 +297,7 @@ get_element_type(c::CellSide) = celltype(c)
 
 abstract type AbstractFaceInfo <: AbstractDomainIndex end
 
+abstract type AbstractFaceInfo <: AbstractDomainIndex end
```
Comment (bmxam) [inline src/mesh/domain.jl:300]: duplicate of line 298?

File: src/cellfunction/cellfunction.jl (hunk 1 above)
Comment (ghislainb) [reply src/cellfunction/cellfunction.jl:565]: done

**Pull request merged**